api/client/start.go

```diff
@@ -112,7 +112,7 @@ func (cli *DockerCli) CmdStart(args ...string) error {
 		})
 
 		// 3. Start the container.
-		if err := cli.client.ContainerStart(context.Background(), container); err != nil {
+		if err := cli.client.ContainerStart(context.Background(), container, ""); err != nil {
```

Comment:
Either way, but this code is just wrong.

Do we need some sort of best practices written up on context? Looking at the code and other PRs, I feel like there is some misunderstanding in effectively and using and propagating context.